core/txpool/legacypool/list.go

```diff
@@ -323,15 +323,20 @@ func (l *list) Add(tx *types.Transaction, priceBump uint64) (bool, *types.Transa
 		if tx.GasFeeCapIntCmp(thresholdFeeCap) < 0 || tx.GasTipCapIntCmp(thresholdTip) < 0 {
 			return false, nil
 		}
-		// Old is being replaced, subtract old cost
-		l.subTotalCost([]*types.Transaction{old})
 	}
 	// Add new tx cost to totalcost
 	cost, overflow := uint256.FromBig(tx.Cost())
 	if overflow {
 		return false, nil
 	}
-	l.totalcost.Add(l.totalcost, cost)
+	total, overflow := new(uint256.Int).AddOverflow(l.totalcost, cost)
+	if overflow {
+		return false, nil
+	}
+	l.totalcost = total
+
+	// Old is being replaced, subtract old cost
+	l.subTotalCost([]*types.Transaction{old})
```

Comment:
The old can be nil. `l.subTotalCost([]*types.Transaction{nil})` will panic, isn't it?

Reply:
yes good catch